src/transformStateWithClones.js

```diff
@@ -8,6 +8,27 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const finalObject = [];
+  let modObject = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(modObject, action.extraData);
+      finalObject.push({ ...modObject });
+    }
+
+    if (action.type === 'removeProperties') {
+      action.keysToRemove.forEach((key) => delete modObject[key]);
+      finalObject.push({ ...modObject });
+    }
+
+    if (action.type === 'clear') {
+      modObject = {};
+      finalObject.push({ ...modObject });
+    }
```

Comment:
Instead of using multiple `if` statements, a `switch` statement would be more appropriate here since you have a limited and known set of action types. This will make the code cleaner and more maintainable.